docs/section21/Rule21-10.md

```diff
@@ -0,0 +1,69 @@
+
+# Boiler - Rule 21-10  
+
+**Rule ID:** 21-10  
+**Rule Description:** When the building uses boilers the hot water pump shall be modeled as riding the pump curve if the hot water system serves less than 120,000 ft^2 otherwise it shall be modeled with a VFD.  
+**Rule Assertion:** B-RMR  
+**Appendix G Section:** Section 21 Boiler  
+**Appendix G Section Reference:** Section G3.1.3.5 Building System-Specific Modeling Requirements for the Baseline model  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:**  
+
+1. B-RMR is modeled with at least one air-side system that is Type-1, 5, 7, 11, or 12
+2. B-RMR is not modeled with purchased heating.
+
+**Manual Check:** None  
+**Evaluation Context:** Building  
+**Data Lookup:** None  
+**Function Call:** None  
+
+**Applicability Checks:**  
+
+1. B-RMR is modeled with at least one air-side system that is Type-1, 5, 7, 11, or 12: `if PLACEHOLDER_AIRSIDE_SYSTEM_RULE-X-XX:`
+2. B-RMR is not modeled with purchased heating: `if NOT Rule-21-1:`
+
+## Rule Logic:  
+
+- For each zone in B_RMR: `for zone_b in B_RMR...zones:`
+
+  - Check if zone has HVAC terminals, get zone terminals: `if zone_b.terminals: zone_terminals_b = zone_b.terminals`
+
+    - For each zone terminal: `for zone_terminal_b in zone_terminals_b:`
+
+      - Check if zone area has not been added to any heating hot water loop: `if NOT zone_flag:`
+
+        - Get HVAC system serving the terminal: `hvac_b = zone_b.terminals.served_by_heating_ventilation_air_conditioning_systems`
+
+          - Get heating system from HVAC system: `heating_system_b = hvac_b.heating_system`
+
+            - Check if heating system has heating hot water loop, get heating hot water loop from heating system: `if heating_system_b.hot_water_loop: hhw_loop_b = heating_system_b.hot_water_loop`
+
+              - Save zone area to heating hot water loop: `hhw_loop_total_area += sum(space.floor_are for space in zone_b.spaces)`
+
+              - Set zone flag as True to avoid double counting the zone area: `zone_flag == TRUE`
+
+- For each pump in B-RMR: `for pump_b in B_RMR.ASHRAE229.pumps:`
+
+  - Check if pump serves heating hot water loop: `if pump_b.loop_or_piping == hhw_loop_b:`
+
+    - Check if heating hot water loop serves less than 120,000 ft^2, set pump type to riding the pump curve: `if hhw_loop_total_area < 120000: pump_type == "RIDING_CURVE"`
+
+    - Else, heating hot water loop serves 120,000 ft^2 or more, set pump type to variable speed: `else: pump_type == "VFD"`
+
+**Rule Assertion:**
+
+- Case 1: For heating hot water loop, if loop serves less than 120,000 ft^2: `if pump_type == "RIDING_CURVE": PASS`
+
```

Comment:
i dont think we have a data element for pump type.
we're using PumSpeedControlOptions for fixed speed/variable speed. and then the FluidLoopFlowControl for fixed flow and variable flow.
I believe Riding_Curve would correspond to Fixed speed, Variable flow and VFD would be Variable Speed, Variable Flow. @mkarpman , please confirm.

Reply:
I think for both VSD and riding pump curve we need the loop to be Variable_Flow (i.e., have 2-way valves modulating with the load). From there, if pump does not have VSD on the motor it will be riding pump curve, and otherwise it will be VSD.

Reply:
Updated.

Reply:
@xingcx I do not see the checks for loop flow control. That still needs to be added. the speed control looks good.

Reply:
Yes, confirmed. But we are checking variable flow in rule 21-11B, so it's not needed here.